marckdown.js

```diff
@@ -1,19 +1,27 @@
 export function render(markdown) {
-  let result = [];
-
   if (typeof markdown === "string") {
     return markdown;
   }
 
-  let child = markdown.content.map(render);
-  console.log(child);
+  let child = markdown.content.map(render).join("");
 
   if (markdown.type === "document") {
-    return child.join("") + "\n\n";
+    return child;
   }
 
   if (markdown.type === "paragraph") {
-    return child.join("") + "\n\n";
+    return child + "\n";
+  }
+
+  if (markdown.type === "heading") {
+    return "# " + child + "\n";
+  }
+
+  if (markdown.type === "bold") {
+    return "**" + child + "**";
+  }
+
+  if (markdown.type === "italic") {
+    return "_" + child + "_";
   }
-  return child;
```

Comment:
`throw new Error('Unexpected node type')` so it'll be an exception if the code execution will ever land here (which should never be the case) rather than giving the wrong answer